src/treelearner/data_parallel_tree_learner.cpp

```diff
@@ -231,6 +231,11 @@ void DataParallelTreeLearner<TREELEARNER_T>::FindBestSplitsFromHistograms(const
   }
 
   // sync global best info
+  auto max_cat_threshold = this->config_->max_cat_threshold;
+  int splitInfoSize = SplitInfo::Size(max_cat_threshold);
+  if (input_buffer_.size() < splitInfoSize * 2) {
+    input_buffer_.resize(splitInfoSize * 2);
+  }
```

Comment:
maybe we can do this in Init function?

Reply:
done!  is input_buffer_ used just as a temporary variable throughout the class?  I don't have a good grasp of the code so I'm not sure if I'm actually making the right fix here.

Reply:
yeah, I just want to minimize the `if ... else' in runtime. 
BTW, should we fix the corresponding code in feature/voting parallel?

Reply:
```
BTW, should we fix the corresponding code in feature/voting parallel?
```
good find, I wasn't aware that was an issue there too

Reply:
done, updated voting parallel version.  It looks like feature parallel already had this logic, but I changed it to use Size method on SplitInfo instead of doing a custom calculation.  Also I modified the data parallel calculation to use std::max to be more in line with the style in the voting parallel class for calculating the input_buffer memory.